### ISSUE TITLE
no dependency pins in conda-package

### ISSUE THREAD

**Issue by k-dominik** [conda-forge/napari-feedstock#7]
it looks like the conda package for napari does not contain any dependency pinning (except Python). Here the `info.json` from the current (0.33) package:
<details>

```json
  "depends": [
    "backcall",
    "cachey",
    "dask",
    "fsspec",
    "imageio",
    "ipykernel",
    "ipython",
    "jupyter_client",
    "napari-plugin-engine",
    "napari-svg",
    "numpy",
    "numpydoc",
    "pluggy",
    "pyopengl",
    "pyside2",
    "python >=3.6",
    "qtconsole",
    "qtpy",
    "scikit-image",
    "scipy",
    "tifffile",
    "vispy",
    "wrapt",
    "zarr"
  ],
```
</details>

napari itself seems to have quite a few pins:

<details>

from `setup.cfg`:

```ini
install_requires =
    cachey>=0.2.1
    dask[array]>=2.1.0
    imageio>=2.5.0
    importlib-metadata>=1.5.0 ; python_version < '3.8'
    ipykernel>=5.1.1
    IPython>=7.7.0
    napari-plugin-engine>=0.1.5
    napari-svg>=0.1.3
    numpy>=1.10.0
    numpydoc>=0.9.2
    Pillow!=7.1.0,!=7.1.1  # not a direct dependency, but 7.1.0 and 7.1.1 broke imageio
    psutil>=5.0
    PyOpenGL>=3.1.0
    PyYAML>=5.1
    qtconsole>=4.5.1
    qtpy>=1.7.0
    scipy>=1.2.0
    tifffile>=2020.2.16
    toolz>=0.10.0
    vispy>=0.6.4
    wrapt>=1.11.1
```

</details>

Should these maybe be added to the recipe here?

**Comment by jni** [comment]
I was advised by conda-forge devs to keep pins to a minimum in conda-forge recipes, as that helps the solver find a solution faster even within the valid possibilities. If this caused an issue (I see in #hci-unihd/covid-if-annotations#30 that it did), then I suggest updating the specific pin that caused an issue.

**Comment by k-dominik** [comment]
> 
> 
> I was advised by conda-forge devs to keep pins to a minimum in conda-forge recipes, as that helps the solver find a solution faster even within the valid possibilities. If this caused an issue (I see in #[hci-unihd/covid-if-annotations#30](https://github.com/hci-unihd/covid-if-annotations/issues/30) that it did), then I suggest updating the specific pin that caused an issue.

Interesting, that is slightly irritating as I would trust the developers of packages the most with dependency versions.

**Comment by jni** [comment]
> Interesting, that is slightly irritating as I would trust the developers of packages the most with dependency versions.

nah we are mostly just winging it! :sweat_smile:

One issue is that pins can become stale, e.g. we had `ipykernel<7` (or similar) for a long time, much longer than the issue existed. Conversely, sometimes a new version of a dependency comes out that breaks things. So I think the philosophy is to leave version pinning to users, who have the latest information.

**Comment by k-dominik** [comment]
> > Interesting, that is slightly irritating as I would trust the developers of packages the most with dependency versions.
> 
> nah we are mostly just winging it! sweat_smile
> 
> One issue is that pins can become stale, e.g. we had `ipykernel<7` (or similar) for a long time, much longer than the issue existed. Conversely, sometimes a new version of a dependency comes out that breaks things. So I think the philosophy is to leave version pinning to users, who have the latest information.

Sure that makes total sense. On the other hand if a user knows well enough which dependencies to pin to get a particular package to run, she could as well open a PR on the corresponding feedstock...

I know the problem of stale pins. On the other hand not doing "conscious" pinning leads to faulty software and with Python that means mostly at run-time... But I'm sure there was a big discussion before concluding that "winging it" is the way to go. This way, at least, incompatibilities surface quickly and can be fixed in the dependent repos :)

**Comment by sofroniewn** [comment]
@jni given that we closed #8 with the merge of #13 can we close this too, or is it slightly different?

**Comment by jni** [comment]
Not different, I think we can close this! 🎉
